build_runner/lib/src/generate/watch_impl.dart

```diff
@@ -340,26 +345,31 @@ class WatchImpl implements BuildState {
           await watcherEnvironment.reader.readAsBytes(rootPackageConfigId));
 
       BuildResult firstBuild;
+      BuildImpl? build;
       try {
-        var build = _build = await BuildImpl.create(
+        build = _build = await BuildImpl.create(
             options, watcherEnvironment, builders, builderConfigOverrides,
             isReleaseBuild: isReleaseMode);
 
         firstBuild = await build
             .run({}, buildDirs: _buildDirs, buildFilters: _buildFilters);
-      } on CannotBuildException {
+      } on CannotBuildException catch (e, s) {
         _terminateCompleter.complete();
 
         firstBuild = BuildResult(BuildStatus.failure, []);
-      } on BuildScriptChangedException {
+        _readyCompleter.completeError(e, s);
+      } on BuildScriptChangedException catch (e, s) {
         _terminateCompleter.complete();
 
         firstBuild = BuildResult(BuildStatus.failure, [],
             failureType: FailureType.buildScriptChanged);
+        _readyCompleter.completeError(e, s);
+      }
+      if (build != null) {
+        assert(_readyCompleter.isCompleted);
```

Comment:
If this assert passes I would expect line 371 to fail.

Should this be `assert(!_readyCompleter.isCompleted)`?